sstables/generation_type.hh

```diff
@@ -35,18 +37,24 @@ public:
 private:
     utils::UUID _value;
 
-    explicit constexpr generation_type(utils::UUID value) noexcept
-        : _value(value) {}
-
 public:
-    generation_type() = delete;
+    // create an invalid sstable identifier
+    generation_type() = default;
 
     // use zero as the timestamp to differentiate from the regular timeuuid,
     // and use the least_sig_bits to encode the value of generation identifier.
     explicit constexpr generation_type(int_t value) noexcept
         : _value(utils::UUID_gen::create_time(std::chrono::milliseconds::zero()), value) {}
+    explicit constexpr generation_type(utils::UUID value) noexcept
+        : _value(value) {}
+    constexpr utils::UUID as_uuid() const noexcept {
+        if (_value.is_null() || _value.timestamp() == 0) {
+            on_internal_error(sstlog, "int generation used as a UUID ");
+        }
+        return _value;
+    }
     constexpr int_t as_int() const noexcept {
-        if (_value.timestamp() != 0) {
+        if (_value.is_null() || _value.timestamp() != 0) {
             on_internal_error(sstlog, "UUID generation used as an int");
```

Comment:
isn't not it misleading to say that a null value is a misusage of generation type? perhaps we should have a assert_non_null() which calls on_internal_error if null and dumps an appropriate error message.

Reply:
sure. i added a check dedicated to the `is_null()` case. but i'd prefer avoid using yet another help. as it is straight forward enough, self-explained, and short.